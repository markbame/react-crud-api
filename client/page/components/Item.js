import React from 'react'
export default ({item, onClick}) =>
<div className="segment">
	<div className="delete" onClick={()=>onClick(item.id, 'del')}>x</div>
	<div>{ item.title }</div>
	<div className="action">
		<span onClick={()=>onClick(item.id, 'dec')}>-</span>
		<span>{ item.count }</span>
		<span onClick={()=>onClick(item.id, 'inc')}>+</span>
	</div>
</div>
