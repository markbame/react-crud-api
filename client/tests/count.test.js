import React from 'react'
import Item from '../page/components/Item'
import makeRequest from '../utils/'

import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('Counter App', () => {
	const items = [
		{title:'big daddy', count: 4, id: '1afczd'},
		{title:'big', count: 0, id: 'acax00'},
		{title:'daddy', count: -1, id: 'acax11'},
		{title:'111', count: 300, id: 'acax11'},
		{title:'1test name', count: 1111, id: 'acax11'}
	]
	it( `List component renders consistently with various inputs`, () => {
		items.map((item) => {
			let component = renderer.create( <Item item={item} onClick={()=>{}}/> )
			let tree = component.toJSON()
				expect(tree).toMatchSnapshot()
		})
	})
})
