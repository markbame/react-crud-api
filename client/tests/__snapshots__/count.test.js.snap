// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App List component renders consistently with various inputs 1`] = `
<div
  className="segment"
>
  <div
    className="delete"
    onClick={[Function]}
  >
    x
  </div>
  <div>
    big daddy
  </div>
  <div
    className="action"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    <span>
      4
    </span>
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
</div>
`;

exports[`Counter App List component renders consistently with various inputs 2`] = `
<div
  className="segment"
>
  <div
    className="delete"
    onClick={[Function]}
  >
    x
  </div>
  <div>
    big
  </div>
  <div
    className="action"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    <span>
      0
    </span>
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
</div>
`;

exports[`Counter App List component renders consistently with various inputs 3`] = `
<div
  className="segment"
>
  <div
    className="delete"
    onClick={[Function]}
  >
    x
  </div>
  <div>
    daddy
  </div>
  <div
    className="action"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    <span>
      -1
    </span>
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
</div>
`;

exports[`Counter App List component renders consistently with various inputs 4`] = `
<div
  className="segment"
>
  <div
    className="delete"
    onClick={[Function]}
  >
    x
  </div>
  <div>
    111
  </div>
  <div
    className="action"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    <span>
      300
    </span>
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
</div>
`;

exports[`Counter App List component renders consistently with various inputs 5`] = `
<div
  className="segment"
>
  <div
    className="delete"
    onClick={[Function]}
  >
    x
  </div>
  <div>
    1test name
  </div>
  <div
    className="action"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    <span>
      1111
    </span>
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
</div>
`;
